### PR TITLE
multiSendMessage method implemented

### PR DESCRIPTION
**Pull request by lumber1000**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 1]
```diff
@@ -441,6 +484,27 @@ private EventID createAndStoreParentEvent(PlaceMessageRequestOrBuilder request,
         }
     }
 
+    private EventID createAndStoreParentEvent(MultiSendRequest request, String actName, Status status) throws IOException {
+        long startTime = System.currentTimeMillis();
+
+        Event event = start()
+                .name(actName + ' ' + request.getMessages(0).getMetadata().getId().getConnectionId().getSessionAlias())
```
Comment (OptimumCode) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:491]: I think we should check the size of the message list before accessing the first element

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 2]
```diff
@@ -441,6 +484,27 @@ private EventID createAndStoreParentEvent(PlaceMessageRequestOrBuilder request,
...
+                .description(request.getDescription())
+                .type(actName)
+                .status(status)
+                .endTimestamp(); // FIXME set properly as is in the last child
```
Comment (OptimumCode) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:495]: You won't be able to do so with distributed events. I think we can remove that FIXME from here

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 3]
```diff
@@ -441,6 +484,27 @@ private EventID createAndStoreParentEvent(PlaceMessageRequestOrBuilder request,
...
+
+        com.exactpro.th2.common.grpc.Event protoEvent = event.toProto(request.getParentEventId());
+        //FIXME process response
+        try {
+            eventBatchMessageRouter.send(EventBatch.newBuilder().addEvents(event.toProto(request.getParentEventId())).build(), "publish", "event");
```
Comment (OptimumCode) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:500]: There is no need in `publish` and `event` attributes as they are automatically added by the router

File: src/main/java/com/exactpro/th2/act/ActHandler.java [hunk 4]
```diff
@@ -578,6 +642,43 @@ private void sendMessage(Message message, EventID parentEventId) throws IOExcept
...
+                    .collect(Collectors.toList())
+            );
+
+            messageRouter.send(batch, SEND_QUEUE_ATTRIBUTE);
+            //TODO remove after solving issue TH2-217
+            //TODO process response
+/*            EventBatch eventBatch = EventBatch.newBuilder()
+                    .addEvents(createSendMessageEvent(parsedMessage, parentEventId))
+                    .build();
+            eventBatchMessageRouter.send(eventBatch, "publish", "event");*/
```
Comment (OptimumCode) [inline src/main/java/com/exactpro/th2/act/ActHandler.java:676]: I believe the ticket TH2-217 is already solved and we can either remove this code completely or uncomment it

**Review by OptimumCode — APPROVED** [review]

File: src/main/java/com/exactpro/th2/act/ActHandler.java (hunk 1 above)
Comment (lumber1000) [reply src/main/java/com/exactpro/th2/act/ActHandler.java:496]: I've added the check to the beginning of `multiSendMessage` method.

Comment (OptimumCode) [reply src/main/java/com/exactpro/th2/act/ActHandler.java:496]: Sorry, missed that

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**